src/dx/publish.rs

```diff
@@ -1,18 +1,59 @@
-//! TODO: Add documentation
+//! Publish module.
+//!
+//! Publish message to a channel.
+//! The publish module contains the [`PublishMessageBuilder`] and [`PublishMessageViaChannelBuilder`].
+//! The [`PublishMessageBuilder`] is used to publish a message to a channel.
+//!
+//! This module is accountable for publishing a message to a channel of the [`PubNub`] network.
+//!
+//! [`PublishMessageBuilder`]: crate::dx::publish::PublishMessageBuilder]
+//! [`PublishMessageViaChannelBuilder`]: crate::dx::publish::PublishMessageViaChannelBuilder]
+//! [`PubNub`]:https://www.pubnub.com/
+
 use crate::{
-    core::{PubNubError, Transport, TransportMethod, TransportRequest},
+    core::{PubNubError, Serialize, Transport, TransportMethod, TransportRequest},
     dx::PubNubClient,
-    Serialize,
 };
 use derive_builder::Builder;
 use std::collections::HashMap;
 use std::ops::Not;
 use urlencoding::encode;
 
-/// TODO: Add documentation
-pub type MessageType = String;
-
-/// TODO: Add documentation
+/// The [`PublishMessageBuilder`] is used to publish a message to a channel.
+///
+/// This struct is used to publish a message to a channel. It is used by the [`publish_message`] method of the [`PubNubClient`].
+/// The [`publish_message`] method is used to publish a message to a channel.
+/// The [`PublishMessageBuilder`] is used to build the request to be sent to the [`PubNub`] network.
+///
+/// # Examples
+/// ```rust
+/// # use pubnub::{PubNubClientBuilder, Keyset};
+///
+/// # #[tokio::main]
+/// # async fn main() -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
What does `#` mean? Why it's not on line 34 nor on lines 44+?

Reply:
It hides that lines from the documentation (to not make a confusion) but these examples are treated as a happy path tests by `cargo test` command.